macOS/DuckDuckGo/Preferences/Model/AboutPreferences.swift

```diff
@@ -28,15 +28,27 @@ final class AboutPreferences: ObservableObject, PreferencesTabOpening {
     private let internalUserDecider: InternalUserDecider
     @Published var isInternalUser: Bool
     private var internalUserCancellable: AnyCancellable?
+    private let featureFlagger: FeatureFlagger
 
-    private init(internalUserDecider: InternalUserDecider) {
+    private init(internalUserDecider: InternalUserDecider,
+                 featureFlagger: FeatureFlagger = NSApp.delegateTyped.featureFlagger) {
+
+        self.featureFlagger = featureFlagger
         self.internalUserDecider = internalUserDecider
         self.isInternalUser = internalUserDecider.isInternalUser
         self.internalUserCancellable = internalUserDecider.isInternalUserPublisher
             .sink { [weak self] in self?.isInternalUser = $0 }
     }
 
 #if SPARKLE
+    var autoRestartAllowed: Bool {
```

Comment:
Maybe something like `allowsRestartWithoutUserConfirmation` would be clearer?

Reply:
On second thought, having `autoRestartAllowed` as inversion of the feature flag value & using it in `guard` statements make the logic hard(er) to follow (double-negativity). I think this is one of the cases where

```
if isFeatureOn {
  doSomething()
} else {
  doSomethingElse() 
}
```

would be much easier to follow (and later easier to delete when feature flag is permanently ON too)

Reply:
Triple-negativity, because the flag itself is negative (`updateWONTAutomaticallyRestartApp`) 😅

Reply:
Using guards for the new logic means the old logic won't show up in the diff as changed, which is what I was looking for.

I understand the comment but I don't personally agree that this would be better:

```swift
guard !updatesWontAutomaticallyRestartAppFeatureFlag else {
    // new logic
}

// old logic
```

I've updated the code to use `useLegacyAutoRestartLogic` instead, so it becomes:

```swift
guard useLegacyAutoRestartLogic else {
    // new logic
}

// old logic
```